enterprise/server/remote_execution/dirtools/dirtools.go

```diff
@@ -641,9 +641,13 @@ func (ff *BatchFileFetcher) batchDownloadFiles(ctx context.Context, req *repb.Ba
 	ff.statsMu.Unlock()
 
 	fileCache := ff.env.GetFileCache()
-	for _, res := range responses {
+	for i, res := range responses {
 		if res.Err != nil {
-			log.CtxInfof(ctx, "Failed to download %s: %s", digest.String(res.Digest), res.Err)
+			digestStr := "<?>"
+			if i >= 0 && i < len(req.Digests) && req.Digests[i] != nil {
+				digestStr = digest.String(req.Digests[i])
```

Comment:
digest.String(res.Digest) is just for logging -- maybe you could make that method return "<nil digest>"?

Reply:
Done - went with `<nil>` since I think that'll make it more obvious in the logs that it's nil